### ISSUE TITLE
Problem with unicode_util_compat on Erlang 20

### ISSUE THREAD

**Issue by zkessin** [benoitc/hackney#416]
When you try to build a release with erlang 20 using Hackney it complains that unicode_util_compat is included twice which prevents the release from being built

**Comment by gmile** [comment]
Seeing this as well.

**Comment by benoitc** [comment]
I'm not reproducing it unfortunately using the latest stable of erlang 20. Can you share a simple case that reproduce it?

**Comment by gmile** [comment]
@benoitc I don't have a quick repro repo right now, but here's a log from a failed Travis build, maybe it will give some clues: https://travis-ci.org/Nebo15/ehealth.api/builds/246116286#L2441

**Comment by skosch** [comment]
FWIW, I just ran into the same issue (using distillery) and was able to get everything working again after a quick `mix deps.clean --all && mix deps.get`.

**Comment by efexen** [comment]
Can confirm just had this with the following setup:

Erlang/OTP 20 [erts-9.0] and Elixir 1.4.4

Failed with:
```
==> Failed to build release:
    
    Duplicated modules: 
    	unicode_util specified in stdlib and unicode_util_compat
```

When running the `mix release` task from Distillery and @skosch recommendation to do `mix deps.clean --all` solved it 👍

**Comment by maartenvanvliet** [comment]
Had the same issue, also with distillery. The @skosch recommendation worked for me as well. 

However, I forgot to add the `deps` directory to my `.dockerignore` so in the following excerpt the `deps` directory was overwritten, retriggering the error. Adding the directory fixed it to the ignore-file fixed it. 

```
COPY mix.exs mix.lock ./
RUN mix deps.clean --all && mix deps.get && mix deps.compile
ADD . .
```

**Comment by benoitc** [comment]
this issue clearly related to the way mix is handling builds using rebar3. can someone provide a reproducible example? I will have a look on it over the we.

**Comment by ericmj** [comment]
Using unicode_util_compat 0.2.0, Elixir 1.5.0 and OTP 20. I can not see the `unicode_util` module being compiled by unicode_util_compat. You can check in `_build/dev/lib/unicode_util_compat/ebin`. Mix creates it's own rebar.config that it passes to rebar before compilation, you can check if it's correct by looking at `_build/dev/lib/unicode_util_compat/mix.rebar.config`, it should not have `pre_hooks` entry, this also looks correct to me.

@benoitc An issue seems to be that unicode_util_compat generates code into src/ which build tools assume is not generated, so it doesn't know it should be cleaned up between compilations. So if you compilemunicode_util_compat on OTP 19 and then switch to OTP 20 the `src/unicode_util.erl` will linger until the dependency is deleted and refetched again, cleaning may not help since it will only clean build artifiacts - not source files.

**Comment by benoitc** [comment]
fixed in 16c29440c6d407cd84ef2d06aeca6f54663fe3f0 . I applied last pr from @ericmj . Thanks again for the patch.
